tests/embedding/test_app.py

```diff
@@ -323,3 +324,64 @@ def test_app_execute_script(embedded_app):
     with pytest.raises(Exception):
         # This will throw an exception since no module named test available
         embedded_app.execute_script("import test")
+
+
```

Comment:
add a test for the pid/tempfile cleanup. You don't need to run launch_gui to use that script, you can just start a dummy subprocess that does a 1 second sleep